### PR TITLE
feat(ui): Implement `<SignUp />` steps

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

- Implements signin steps
- ~Adds `<FieldEnabled />` component to conditionally render fields~
- ~Adds `<FieldAdapter />` component to check if attr is enabled or required~

https://linear.app/clerk/issue/SDKI-37/implement-signup-steps
https://linear.app/clerk/issue/SDKI-52/implement-required-logic

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/ui/src/primitives/card.tsx [hunk 1]
```diff
@@ -132,7 +132,7 @@ export const FooterAction = React.forwardRef<HTMLDivElement, React.HTMLAttribute
       <div
         ref={forwardedRef}
         {...props}
-        className={cn('px-6 py-4', className)}
+        className={cn('px-6 py-4 border-b border-gray-a6 ', className)}
```
Comment (alexcarpenter) [inline packages/ui/src/primitives/card.tsx:135]: Moves the border to the conditionally rendered element for when the FooterAction isn't used 
Screenshot 2024-06-11 at 4 12 47 PM

File: packages/ui/src/components/sign-in/sign-in.tsx [hunk 2]
```diff
@@ -7,12 +7,12 @@ import { OTPField } from '~/common/otp-field';
 import { PasswordField } from '~/common/password-field';
 import { PhoneNumberField } from '~/common/phone-number-field';
 import { PROVIDERS } from '~/constants';
-import { getEnabledSocialConnectionsFromEnvironment } from '~/hooks/getEnabledSocialConnectionsFromEnvironment';
 import { Button } from '~/primitives/button';
 import * as Card from '~/primitives/card';
 import * as Connection from '~/primitives/connection';
 import * as Icon from '~/primitives/icon';
 import { Seperator } from '~/primitives/seperator';
+import { getEnabledSocialConnectionsFromEnvironment } from '~/utils/getEnabledSocialConnectionsFromEnvironment';
```
Comment (alexcarpenter) [inline packages/ui/src/components/sign-in/sign-in.tsx:15]: Not technically a hook so moved to utils.

File: packages/ui/src/components/sign-up/sign-up.tsx [hunk 3]
```diff
@@ -1,34 +1,51 @@
...
   return (
     <SignUp.Root>
-      <Common.Loading>
-        {isGlobalLoading => {
-          return (
-            <SignUp.Step
-              data-clerk
-              name='start'
-            >
+      <SignUpComponentLoaded />
```
Comment (alexcarpenter) [inline packages/ui/src/components/sign-up/sign-up.tsx:25]: Move to loaded component so we can know that clerk has loaded via elements `<SignUp.Root>` to interface with useClerk hook

File: packages/ui/src/components/sign-up/sign-up.tsx [hunk 4]
```diff
@@ -1,34 +1,51 @@
...
+    (clerk as any)?.__unstable__environment as EnvironmentResource,
+  );
+  const locationBasedCountryIso = (clerk as any)?.__internal_country;
+  const attributes = ((clerk as any)?.__unstable__environment as EnvironmentResource)?.userSettings.attributes;
+  const { enabled: firstNameEnabled, required: firstNameRequired } = attributes['first_name'];
+  const { enabled: lastNameEnabled, required: lastNameRequired } = attributes['last_name'];
+  const { enabled: usernameEnabled, required: usernameRequired } = attributes['username'];
+  const { enabled: phoneNumberEnabled, required: phoneNumberRequired } = attributes['phone_number'];
+  const { enabled: emailAddressEnabled, required: emailAddressRequired } = attributes['email_address'];
+  const { enabled: passwordEnabled, required: passwordRequired } = attributes['password'];
```
Comment (alexcarpenter) [inline packages/ui/src/components/sign-up/sign-up.tsx:42]: Explored a couple of different components to expose this data, but ultimately decided to keep things simple and just access the data here directly. Theres likely an abstraction that can be made here, but keeping it simple atm.

**Review by tmilewski — APPROVED** [review]

**Pull request merged**